Modules/itertoolsmodule.c

```diff
@@ -233,6 +232,12 @@ batched_next(batchedobject *bo)
         Py_DECREF(result);
         return NULL;
     }
+    if (bo->strict) {
+        Py_CLEAR(bo->it);
+        Py_DECREF(result);
+        PyErr_SetString(PyExc_ValueError, "batched(): incomplete batch");
```

Comment:
```suggestion
        PyErr_SetString(PyExc_ValueError, "length of input iterable to batched() is not divisible by n");
```

Reply:
Thanks for the suggestion but I prefer the current message.  From a user's point of view, the issue is not about divisibility, which is just a mathematical artifact.  The issue is that a batch is incomplete.   In the `grouper` recipe, the handling of incomplete batches is controlled with the `incomplete` keyword.